lib/almodovar/single_resource.rb

```diff
@@ -1,66 +1,75 @@
 module Almodovar
   class SingleResource
     include HttpAccessor
-    
+
     undef_method :id if instance_methods.include?("id")
     undef_method :type if instance_methods.include?("type")
-        
+
     def initialize(url, auth, xml = nil, options = {})
       @url = url
       @auth = auth
       @xml = xml
       @options = options
     end
-  
+
     def update(attrs = {})
       raise ArgumentError.new("You must specify one only root element which is the type of resource (e.g. `:project => { :name => 'Wadus' }` instead of just `:name => 'Wadus'`)") if attrs.size > 1
       root, body = attrs.first
       response = http.put(@url, body.to_xml(:root => root), "Content-Type" => "application/xml")
       check_errors(response, @url)
       @xml = Nokogiri::XML.parse(response.body).root
     end
-  
+
     def delete
       check_errors(http.delete(@url), @url)
     end
-  
+
     def url
       @url ||= xml.at_xpath("./link[@rel='self']").try(:[], "href")
     end
-  
+
+    def to_hash
+      xml.delete("type")
```

Comment:
Do we need that `xml.delete("type")` here? This method is supported just for `SingleResource`.

Apart from that, that call to `delete` could be mutating the internal XML representation of the resource, is that OK?